Adm1266 add blackbox support

tachicialex opened the pull request:
Adm1266 will write all input/output status in 64 bytes (blackbox).
This series adds support for reading the states through debugfs and blackbox
through nvmem provider api.

commodo commented on drivers/hwmon/pmbus/adm1266.c:
```diff
@@ -221,6 +224,48 @@ static inline int adm1266_config_gpio(struct adm1266_data *data)
+	return i2c_smbus_write_word_data(data->client, ADM1266_GO_COMMAND, reg);
+}
+
+DEFINE_SIMPLE_ATTRIBUTE(go_command_fops, NULL, adm1266_set_go_command_op,
```
better make this `DEFINE_DEBUGFS_ATTRIBUTE` and use `debugfs_create_file_unsafe()` instead of `debugfs_create_file()`
there are people going around the kernel hunting for these sort of changes;

commodo commented on drivers/hwmon/pmbus/adm1266.c:
```diff
@@ -264,6 +305,120 @@ static void adm1266_debug_init(struct adm1266_data *data)
+
+	for (i = 0; i < data->nvmem_config.ncells; i++) {
+		mem_cell = &adm1266_nvmem_cells[i];
+		if (adm1266_cell_is_accessed(mem_cell, offset, bytes)) {
```
just so that reduce some levels of indentation you can maek this:
```
if (!adm1266_cell_is_accessed(mem_cell, offset, bytes))
     continue;
```

commodo commented:
a few minor notes;
rest looks good

[![LGTM](https://lgtm.ro/wp-content/uploads/2019/05/1ee5ol-1.jpg)](https://lgtm.ro/wp-content/uploads/2019/05/1ee5ol-1.jpg)

tachicialex commented:
Change-log v1 -> v2:
- replaced DEFINE_SIMPLE_ATTRIBUTE with DEFINE_DEBUGFS_ATTRIBUTE
- switched  debugfs_create_file() to debugfs_create_file_unsafe()
- used continue; statement at adm1266_cell_is_accessed(mem_cell, offset, bytes));
